### ISSUE TITLE
bug in update of a document

### ISSUE THREAD

**Issue by chrishna1** [typesense/typesense-go#90]
## Description

Updating a document return non nil error even when update is successful.

## Steps to reproduce

```go
package main

import (
	"fmt"
	"os"

	"github.com/typesense/typesense-go/typesense"
	"github.com/typesense/typesense-go/typesense/api"
)

func main() {
	client := typesense.NewClient(
		typesense.WithServer(os.Getenv("TS_HOST")),
		typesense.WithAPIKey(os.Getenv("TS_APIKEY")))

	// resDrop, err := client.Collection("companies").Delete()

	// if err != nil {
	// 	fmt.Println("Some error occured", err)
	// 	return
	// }

	// fmt.Printf("collection dropped successfully!!, res: %#v \n\n", resDrop)

	dsf := "num_employees"
	boolp := true

	schema := &api.CollectionSchema{
		Name: "companies",
		Fields: []api.Field{
			{
				Name: "company_name",
				Type: "string",
			},
			{
				Name: "num_employees",
				Type: "int32",
			},
			{
				Name:  "country",
				Type:  "string",
				Facet: &boolp,
			},
		},
		DefaultSortingField: &dsf,
	}

	resCreate, err := client.Collections().Create(schema)

	if err != nil {
		fmt.Println("Some error occured", err)
		return
	}

	fmt.Printf("collection created successfully!!, res: %#v \n\n", resCreate)

	document := struct {
		ID           string `json:"id"`
		CompanyName  string `json:"company_name,omitempty"`
		NumEmployees int    `json:"num_employees,omitempty"`
		Country      string `json:"country,omitempty"`
	}{
		ID:           "123",
		CompanyName:  "Stark Industries",
		NumEmployees: 5215,
		Country:      "USA",
	}

	resUpsert, err := client.Collection("companies").Documents().Upsert(document)

	if err != nil {
		fmt.Println("Some error occured", err)
		return
	}

	fmt.Printf("document indexed successfuly!!, res: %#v \n\n", resUpsert)

	newDocument := struct {
		CompanyName  string `json:"company_name"`
		NumEmployees int    `json:"num_employees"`
	}{
		CompanyName:  "Stark Industries hoohaahaa",
		NumEmployees: 5500,
	}

	resUpdate, err := client.Collection("companies").Document("123").Update(newDocument)

	if err != nil {
		fmt.Println("Some error occured", err)
		return
	}

	fmt.Printf("document updated successfully!!, res: %#v \n\n", resUpdate)
}
```

## Output

> collection created successfully!!, res: &api.CollectionResponse{CollectionSchema:api.CollectionSchema{DefaultSortingField:(*string)(0xc00009a0c0), Fields:[]api.Field{api.Field{Facet:(*bool)(0xc0000a0038), Index:(*bool)(0xc0000a0039), Name:"company_name", Optional:(*bool)(0xc0000a003a), Type:"string"}, api.Field{Facet:(*bool)(0xc0000a0086), Index:(*bool)(0xc0000a0087), Name:"num_employees", Optional:(*bool)(0xc0000a0088), Type:"int32"}, api.Field{Facet:(*bool)(0xc0000a008e), Index:(*bool)(0xc0000a008f), Name:"country", Optional:(*bool)(0xc0000a00a7), Type:"string"}}, Name:"companies", SymbolsToIndex:[]string(nil), TokenSeparators:[]string(nil)}, CreatedAt:1637679184, NumDocuments:0, NumMemoryShards:4} 

> document indexed successfuly!!, res: map[string]interface {}{"company_name":"Stark Industries", "country":"USA", "id":"123", "num_employees":5215} 

> Some error occured status: 201 response: {"company_name":"Stark Industries hoohaahaa","country":"USA","id":"123","num_employees":5500}

## Expected Behavior

err should be nil as update is successful

## Actual Behavior

err is non nil

## Metadata

**Typsense Version**:
docker - typesense:0.21.0
go-client - v0.3.0

**OS**:
Ubuntu 21.10

**Comment by harisarang** [comment]
Hi @chrishna1, thank you for catching this and thank you for the PR. These files are however auto-generated and so we don't want to edit them directly. Instead, I've made changes to the source [api spec](https://github.com/typesense/typesense-api-spec/blob/master/openapi.yml) that's used to generate this file,

```bash
go get github.com/deepmap/oapi-codegen/cmd/oapi-codegen@v1.9.0 #or
go install github.com/deepmap/oapi-codegen/cmd/oapi-codegen@v1.9.0

go generate
```

Now, it should regenerate this file. For more info, look into readme of the repository. Do you want to update the PR based on that?

**Comment by chrishna1** [comment]
Sure. I'll update it.

**Comment by chrishna1** [comment]
fixed in #91
